src/main/java/net/sf/mzmine/modules/peaklistmethods/io/gnpsexport/GNPSExportTask.java

```diff
@@ -32,272 +32,240 @@
+
+      // Open file
+      FileWriter writer;
+      try {
+        writer = new FileWriter(curFile);
+      } catch (Exception e) {
+        setStatus(TaskStatus.ERROR);
+        setErrorMessage("Could not open file " + curFile 
+            + " for writing.");
+        return;
+      }
+
+      try {
+        export(peakList, writer, curFile);
+      } catch (IOException e) {
+        setStatus(TaskStatus.ERROR);
+        setErrorMessage("Error while writing into file " + curFile +
+            ": " + e.getMessage());
+        return;
+      }
+
+      // Cancel?
+      if (isCanceled()) {
+        return;
+      }
+
+      // Close file
+      try {
+        writer.close();
+      } catch (Exception e) {
+        setStatus(TaskStatus.ERROR);
+        setErrorMessage("Could not close file " + curFile);
+        return;
+      }
+
+      // If peak list substitution pattern wasn't found, 
+      // treat one peak list only
+      if (!substitute)
+        break;
     }
 
-	@Override
-	public double getFinishedPercentage() {
-		return 0;
-	}
-
-	@Override
-	public void run() {
-		setStatus(TaskStatus.PROCESSING);
-
-        // Shall export several files?
-        boolean substitute = fileName.getPath().contains(plNamePattern);
-
-        // Process peak lists
-        for (PeakList peakList: peakLists) {
-
-            // Filename
-            File curFile = fileName;
-            if (substitute) {
-                // Cleanup from illegal filename characters
-                String cleanPlName = peakList.getName().replaceAll(
-                        "[^a-zA-Z0-9.-]", "_");
-                // Substitute
-                String newFilename = fileName.getPath().replaceAll(
-                        Pattern.quote(plNamePattern), cleanPlName);
-                curFile = new File(newFilename);
-            }
+    if (getStatus() == TaskStatus.PROCESSING)
+      setStatus(TaskStatus.FINISHED);
+  }
 
-            // Open file
-            FileWriter writer;
-            try {
-                writer = new FileWriter(curFile);
-            } catch (Exception e) {
-                setStatus(TaskStatus.ERROR);
-                setErrorMessage("Could not open file " + curFile 
-                        + " for writing.");
-                return;
-            }
+  private void export(PeakList peakList, FileWriter writer, File curFile)
+      throws IOException
+  {
+    final String newLine = System.lineSeparator();
 
-            try {
-            	export(peakList, writer, curFile);
-            } catch (IOException e) {
-                setStatus(TaskStatus.ERROR);
-                setErrorMessage("Error while writing into file " + curFile +
-                        ": " + e.getMessage());
-                return;
-            }
+    int currentID = 1;
+    for (int i = 0; i < peakList.getRows().length; i++) {
+      PeakListRow row = peakList.getRows()[i];
+      String rowID = Integer.toString(row.getID());
+      if (currentID != row.getID()) {
```

Comment:
These are necessary for existing file readers and backwards compatibility in gnps